internal/clients/ec2/ec2_client.go

```diff
@@ -0,0 +1,38 @@
+package ec2
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/sts/types"
+)
+
+func CreateNewEC2Client(crd *types.Credentials) (*ec2.Client, error) {
+	newCfg, err := config.LoadDefaultConfig(context.Background(), config.WithCredentialsProvider(
+		credentials.NewStaticCredentialsProvider(*crd.AccessKeyId, *crd.SecretAccessKey, *crd.SessionToken),
+	))
+	if err != nil {
+		e := ec2.NewFromConfig(newCfg)
+		return e, err
+	}
+	return nil, err
+}
+
+func ListRegions(e *ec2.Client) (string, error) {
+	var result string
+
+	resultRegions, err := e.DescribeRegions(context.TODO(), &ec2.DescribeRegionsInput{})
+	if err != nil {
+		return "", err
+	}
+
+	for _, instance := range resultRegions.Regions {
+		instanceJSON, _ := json.MarshalIndent(instance, "", "  ")
+		result = result + string(instanceJSON)
+	}
+
+	return result, err
+}
```

Comment:
Yeah, I renamed clouds/ to clients/ so it matches your package. And I am changing my mind about the package name - let's use aws/ instead of ec2/ because we can put both ec2 and cloudwatch APIs into the same connection (client interface).

https://github.com/RHEnVision/provisioning-backend/pull/13

Reply:
Correction: let's actually use ec2 and cw separately, there is a big difference between the two - cw is connected for the whole application globally while ec2 we will create on the fly. So keep ec2 here, I will rename it.

Reply:
I think we can't use the same connection, cloudwatch will have different credentials then the ec2 connection IMHO :thinking:

Reply:
Is it something we should add to the config? like creds for cloudwatch and for others?